### ISSUE TITLE
python3.6: re has no member M

### ISSUE THREAD

**Issue by asottile** [PyCQA/pylint#1282]
### Steps to reproduce

Example python file:

```python
import re
re.compile('foo\nbar', re.M)
```

Using a pylintrc similar to this one:

```ini
[MESSAGES CONTROL]
disable=missing-docstring

[REPORTS]
output-format=colorized
reports=no
```

### Current behavior

```
$ pylint example.py 
************* Module example
E:  2,23: Module 're' has no 'M' member (no-member)
```

### Expected behavior

No output

### pylint --version output

```
$ pylint --version
pylint 1.6.4, 
astroid 1.4.9
Python 3.6.0 (default, Jan 13 2017, 20:56:47) 
[GCC 5.4.0 20160609]
```

**Comment by rogalski** [comment]
Looks like it was caused by this change: https://github.com/python/cpython/commit/deec4d0aea8eb0225c4fe3999815a2cb6c0ad20c . Since global namespace is populated auto-magically, pylint cannot infer them.

Looks like a job for astroid.brain.

**Comment by PCManticore** [comment]
Thanks @rogalski. That's a shame, the previous code was much better. Definitely an astroid.brain job.

**Comment by hajs** [comment]
The following module brain_re.py works for me. Is this the correct way?

```python
# Licensed under the LGPL: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
# For details: https://github.com/PyCQA/astroid/blob/master/COPYING.LESSER
import sys
import astroid


PY36 = sys.version_info >= (3, 6)

if PY36:
    # Since Python 3.6 there is the RegexFlag enum 
    # where every entry will be exposed via updating globals()
    
    def _re_transform():
        return astroid.parse('''
        import sre_compile
        ASCII = sre_compile.SRE_FLAG_ASCII
        IGNORECASE = sre_compile.SRE_FLAG_IGNORECASE
        LOCALE = sre_compile.SRE_FLAG_LOCALE
        UNICODE = sre_compile.SRE_FLAG_UNICODE
        MULTILINE = sre_compile.SRE_FLAG_MULTILINE
        DOTALL = sre_compile.SRE_FLAG_DOTALL
        VERBOSE = sre_compile.SRE_FLAG_VERBOSE
        A = ASCII
        I = IGNORECASE
        L = LOCALE
        U = UNICODE
        M = MULTILINE
        S = DOTALL
        X = VERBOSE
        TEMPLATE = sre_compile.SRE_FLAG_TEMPLATE
        T = TEMPLATE
        DEBUG = sre_compile.SRE_FLAG_DEBUG
        ''')
    
    
    astroid.register_module_extender(astroid.MANAGER, 're', _re_transform)
```

**Comment by rogalski** [comment]
@hajs It should do the trick.

@PCManticore I was wondering whether astroid can we actually understand `globals().update(some_dict)` calls, when `some_dict` can be inferred (which should be the case here).

**Comment by PCManticore** [comment]
This looks good. Some tests will help as well! 

@rogalski it could work, in theory. It depends if the arguments to the update call can be inferred and the scope of the globals call. What I mean by this is that statements up to the globals update should not see the newly modified values, but this might be tricky, because we can't know the flow of the program that leads to that update. Anyway, this is a potential uninteresting edge case, but we should approach it in an astroid issue.

**Comment by rogalski** [comment]
@hajs do you want to create a proper PR in astroid?

**Comment by lucc** [comment]
I have no idea about the internals of pylint but the relevant constants are listed in `re.__all__`. @hajs does it make more sense to read names form `re.__all__` than hardcoding them like you suggested?

(I actually came here to report the same bug, just for `re.IGNORECASE`.)

**Comment by rogalski** [comment]
FYI, I decided to do PR myself: PyCQA/astroid#407

**Comment by PCManticore** [comment]
Thank you @hajs and @rogalski This is now fixed in astroid's master.
